src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -4,113 +4,124 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Microsoft.Framework.Configuration.Internal;
 using Microsoft.Framework.Internal;
 
 namespace Microsoft.Framework.Configuration
 {
-    public class ConfigurationSection : IConfiguration
+    public class ConfigurationSection : ConfigurationBase, IConfigurationSection
     {
-        private readonly IList<IConfigurationSource> _sources = new List<IConfigurationSource>();
+        private string _key;
 
-        public ConfigurationSection(IList<IConfigurationSource> sources)
+        public ConfigurationSection([NotNull] IList<IConfigurationSource> sources)
+            : base(sources)
         {
-            _sources = sources;
+            _key = string.Empty;
+            Value = null;
         }
-
-        public string this[string key]
+        public ConfigurationSection([NotNull] IList<IConfigurationSource> sources, [NotNull] string key)
+            : base (sources)
         {
-            get
-            {
-                return Get(key);
-            }
-            set
+            if (string.IsNullOrEmpty(key))
             {
-                Set(key, value);
+                throw new InvalidOperationException(Resources.Error_EmptyKey);
             }
+
+            _key = key;
+            Value = base[_key];
         }
 
-        public IEnumerable<IConfigurationSource> Sources
+        public string Key
         {
             get
             {
-                return _sources;
+                return _key;
             }
         }
 
-        public string Get([NotNull] string key)
+        public string Value
```

Comment:
I don`t think we should have an automatic property here. This should be getting and setting the value that corresponds to the current key from the sources.